qt/widgets/common/inc/MantidQtWidgets/Common/QSettingsHelper.h

```diff
@@ -0,0 +1,96 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_QSETTINGSHELPER_H
+#define MANTID_QSETTINGSHELPER_H
+
+#include <QMetaType>
+#include <QSettings>
+#include <QString>
+#include <QStringList>
+#include <QVariant>
+#include <map>
+#include <string>
+#include <typeinfo>
+
+namespace MantidQt {
+namespace MantidWidgets {
+namespace QSettingsHelper {
+
+/** Loads an individual setting from disk
+ *
+ * @param settingGroup The name of the setting group
+ * @param settingName The name of the setting
+ * @return The value stored for the requested setting
+ *
+ */
+template <typename T>
+T getSetting(std::string const &settingGroup, std::string const &settingName) {
+  QSettings settings;
+  settings.beginGroup(QString::fromStdString(settingGroup));
+  auto const settingValue = settings.value(QString::fromStdString(settingName));
+  settings.endGroup();
+
+  return settingValue.value<T>();
+}
+
+/** Loads a map of settings with the same type. This comparison is required
+ *  as QVariant types are not properly encoded inini files
+ *
+ * @param settingGroup The name of the setting group
+ * @return A map of the values stored for all settings matching the given type
+ *
+ */
+template <typename T>
+std::map<std::string, T> getSettingsAsMap(std::string const &settingGroup) {
+  std::map<std::string, T> settingsMap;
+  QSettings settings;
+  settings.beginGroup(QString::fromStdString(settingGroup));
+  QStringList settingNames = settings.allKeys();
+  std::string templateTypeName = typeid(T).name();
+  for (auto &settingName : settingNames) {
+    std::string settingTypeName;
+    if (settingName.endsWith("/type")) {
+      settingTypeName = settings.value(settingName).toString().toStdString();
```

Comment:
`settingTypeName` looks like it could be defined here, avoiding creating the string object if it does not end in `"/type"`